grammar.js

```diff
@@ -95,13 +98,30 @@ module.exports = grammar(require('tree-sitter-javascript/grammar'), {
     [$._property_definition_list, $._property_name, $._expression],
 
     [$.call_signature, $.function_type],
-    [$.constructor_type, $.call_signature]
+    [$.constructor_type, $.call_signature],
+
+    [$._module, $._entity_name],
+
+    [$._expression, $.import_alias],
+
+    [$._expression, $.method_definition],
+
+    [$.reserved_identifier, $.module],
+    [$.reserved_identifier, $.reserved_identifier],
+    [$.reserved_identifier, $.ambient_declaration],
+    [$.reserved_identifier, $.interface_declaration],
+    [$.reserved_identifier, $.internal_module],
+    [$.reserved_identifier, $.abstract_method_definition],
+    [$.reserved_identifier, $.public_field_definition, $.abstract_method_definition],
+
+    [$.reserved_identifier, $.type_alias_declaration]
```

Comment:
Most conflicts added are related to the addition of a `reserved_identifier` rule for `declare`, `namespace`, `type` and other TypeScript reserved keywords that can also be used as identifiers (e.g. `var type = 'a';`). @maxbrunsfeld I wasn't successful in solving ambiguity conflicts without adding these conflict resolutions, which expands the size of the parser. Please let me know if there are other strategies that we can try if this is problematic 🙇